spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -152,21 +164,52 @@ public void setExecutor(Executor executor) {
 		this.executorExplicitlySet = true;
 	}
 
+	/**
+	 * Rebalance the capacity of cached locks.
+	 * @param capacity The capacity of cached lock
+	 * @since 5.5.6
+	 */
+	public void setCapacity(long capacity) {
+		synchronized (this.locks) {
+			final int locksSize = this.locks.size();
+			final int newLockMapSize = (int) capacity;
+			if (locksSize > newLockMapSize) {
+				final int removeCnt = locksSize - newLockMapSize;
+				Iterator<Entry<String, RedisLock>> iterator = this.locks.entrySet().iterator();
+				for (int i = 0; i < removeCnt; i++) {
+					iterator.next();
+					iterator.remove();
```

Comment:
I don't think that someone is going to modify the `capacity` at runtime.
Move over we consult with it in that `removeEldestEntry` at runtime any way.
So, new value is going to be an effect and therefore obsolete entries are going to be removed on the next `put()`.

Reply:
Thanks for the review.
Fixed not to rebalance in setter